### PR TITLE
ref(js): Improve structure of OrganizationLayout

### PR DESCRIPTION
**Pull request by evanpurkhiser**
- Renames OrganizationDetails -> OrganizationLayout. This makes a lot more sense since this component is responsible for the general page layout / structure.

 - Renames OrganizationLayout -> OrganizationContainer. This makes a lot more sense since the component should be responsible for only waiting to load the organization into context

This PR makes adjustments to both of these components to make the responsibilities clear to follow the above two points.

### CODE REVIEW

File: static/app/views/organizationContainer.tsx [hunk 1]
```diff
@@ -13,19 +11,16 @@ import {space} from 'sentry/styles/space';
...
 
-interface OrganizationLayoutProps {
-  children: React.ReactNode;
+interface Props {
+  children: JSX.Element;
 }
 
-const OrganizationHeader = HookOrDefault({
-  hookName: 'component:organization-header',
-});
```
Comment (evanpurkhiser) [inline static/app/views/organizationContainer.tsx:22]: This moves to the new `OrganizationLayout`

File: static/app/views/organizationContainer.tsx [hunk 2]
```diff
@@ -49,18 +44,11 @@ function OrganizationLayout({children}: OrganizationLayoutProps) {
     return <ErrorWrapper>{errorBody}</ErrorWrapper>;
   }
 
-  return (
-    <SentryDocumentTitle noSuffix title={organization?.name ?? 'Sentry'}>
-      <div className="app">
-        {organization && <OrganizationHeader organization={organization} />}
-        {children}
-      </div>
-    </SentryDocumentTitle>
```
Comment (evanpurkhiser) [inline static/app/views/organizationContainer.tsx:58]: This moves to the new `OrganizationLayout`

File: static/app/views/organizationContainer.tsx [hunk 3]
```diff
@@ -49,18 +44,11 @@ function OrganizationLayout({children}: OrganizationLayoutProps) {
...
-  );
+  return children;
 }
 
 const ErrorWrapper = styled('div')`
   padding: ${space(3)};
 `;
 
-export default OrganizationLayout;
+export default OrganizationContainer;
```
Comment (evanpurkhiser) [inline static/app/views/organizationContainer.tsx:54]: This component is now only responsible for ensuring the organization is loaded.

File: static/app/views/organizationLayout/index.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+  // XXX(epurkhiser): The OrganizationContainer is responsible for ensuring the
+  // oganization is loaded before rendering children. Organization may not be
+  // loaded yet when this first renders.
+  const organization = useOrganization({allowNull: true});
+
+  return (
+    <SentryDocumentTitle noSuffix title={organization?.name ?? 'Sentry'}>
+      <OrganizationContainer>
+        <div className="app">
```
Comment (evanpurkhiser) [inline static/app/views/organizationLayout/index.tsx:30]: Unclear if we actually need this class or not.

**Review by scttcper — APPROVED** [review]

**Pull request merged**